scylla-cql/src/frame/frame_errors.rs

```diff
@@ -87,7 +85,24 @@ pub enum SetKeyspaceParseError {
 }
 
 /// An error type returned when deserialization of
-/// `RESULT::Schema_change` response fails.
+/// `EVENT` response fails.
+#[non_exhaustive]
+#[derive(Error, Debug)]
+pub enum CqlEventParseError {
+    #[error("Malformed event type string: {0}")]
+    EventTypeParseError(LowLevelDeserializationError),
+    #[error("Unknown event type: {0}")]
+    UnknownEventType(String),
+    #[error("Failed to deserialize schema change event: {0}")]
+    SchemaChangeEventParseError(#[from] SchemaChangeEventParseError),
+    #[error("Failed to deserialize topology change event: {0}")]
+    TopologyChangeEventParseError(ClusterChangeEventParseError),
+    #[error("Failed to deserialize status change event: {0}")]
+    StatusChangeEventParseError(ClusterChangeEventParseError),
```

Comment:
Why did you decide to unify Topology and Status errors into common Cluster error? Could you summarise the reasons?

Reply:
I think not all variants of ClusterChangeEventParseError apply to both Topology and Status.

Reply:
All of the variants are shared between status and topology changes. I decided to unify them to reduce the copy-paste.